src/request_byte_tree_node.h

```diff
@@ -0,0 +1,173 @@
+#ifndef REQUEST_BYTE_TREE_NODE_H
+#define REQUEST_BYTE_TREE_NODE_H
+
+#include <map>
+#include <memory>
+#include <optional>
+
+#include "selene.h"
+
+using namespace std;
+using namespace sel;
+
+/**
+ * In order to quickly find a request from the simulation that fits the current
+ * request coming from the application, all possible requests are defined in
+ * in a tree structure of bytes. Each element in the tree (i.e. each byte) contains
+ * pointers to possible next bytes or the actual response
+ * 
+ * This class represents one element (byte) in this tree.
+ * 
+ * For example there are the following requests defined in a simulation file
+ * * 22 F1 90 
+ * * 22 30 98
+ * * 11 01
+ * * 36 XX *
+ * * 31 XX 12
+ * * 31 01 12
+ * 
+ * Then the tree will look like
+ * - 22
+ *   - F1
+ *     - 90 -> response1
+ *   - 30
+ *     - 98 -> response2
+ * - 11
+ *   - 01 -> response3
+ * - 36
+ *   - XX
+ *     - * -> response4
+ * - 31
+ *   - XX
+ *     - 12 -> response5
+ *   - 01
+ *     - 12 -> response6
+ * 
+ * When the application sends request 31 01 12, the simulation can quickly move through
+ * the tree to find the matching response "response5"
+ *
+ */
+template<class T>
```

Comment:
Should I leave that as template class for flexibility or should I use an explicit type?
Currently I only need one type.

Reply:
Don't know. Do you see extension possibilities for other types?